pkg/mount/gpt/gpt.go

```diff
@@ -305,9 +306,14 @@ func writeGPT(w io.WriterAt, g *GPT) error {
 	if err := binary.Write(&b, binary.LittleEndian, &g.Header); err != nil {
 		return err
 	}
-	h = make([]byte, g.HeaderSize)
-	copy(h, b.Bytes())
-	g.CRC = crc32.ChecksumIEEE(h[:])
+
+	h = make([]byte, BlockSize)
+	// Copy the non-reserved portions of the header. The remaining (reserved)
+	// bytes do not need to be zeroed out because the default zero value for
+	// a byte is 0.
+	copy(h[0:g.HeaderSize], b.Bytes())
```

Comment:
Seems to be all we need here is
https://go.dev/play/p/cbus5kmTjge
In that form, there are no magic numbers, save for the 512.
It's a shame the GPT did not include a block size as an element, that would have worked, but they did not.
But I'd rather do something like this than the _ form above if you are ok with that.

Reply:
Sounds good to me.  g.HeaderSize must still be used when calculating the checksum (otherwise checksum differs).   I also used BlockSize instead of 512 when defining 'block' so that this assumption is more discoverable in the future  (when we refactor, it'll be obvious that we have to find all references to 'BlockSize' which is defined high up in the file... it will be less obvious to think about finding instances of '512', IMHO).

How does this look to you?